Setting-Up-Ubuntu-For-Programming.md

```diff
@@ -0,0 +1,219 @@
+# Setting up Ubuntu for Programming
+
+> Note: We will be referring to Ubuntu 16.04 LTS (Xenial Xerus) in this article.
+
+1. Download and Install Ubuntu Desktop
+2. Customising Ubuntu
+3. Jazzing up the Terminal
+4. Installing DevTools and modern web browsers
+5. Downloading Text Editors and IDEs
+6. Alternatives to popular Windows and Mac software
+
+## Download and Install Ubuntu Desktop
+
+You can download the latest version of Ubuntu Desktop [here](http://www.ubuntu.com/download/desktop).
+If you need any help with installing the operating system [this](http://www.ubuntu.com/download/desktop/install-ubuntu-desktop) tutorial by Ubuntu will help you out. If you want to dual boot your computer with Windows and Ubuntu checkout [this](https://help.ubuntu.com/community/WindowsDualBoot) helpful article.
+
+Once Ubuntu is installed, open the terminal and type `sudo apt-get update && sudo apt-get upgrade` to update your packages.
+
+## Customising Ubuntu
+
+#### `.deb` Package Installer
+
+Install gdebi (`.deb` package installer) by typing `sudo apt-get install gdebi`.
```

Comment:
`gdebi` is used to install `.deb` packages when you double click them. This is non terminal, the most seen behavior for newbies. Even I don't dpkg unless it is from a tutorial that I'm copy/pasting.

With regards to apt vs apt-get I don't care much. I started using apt because the tutorials I have seen for 16.04 have been using apt and I like it, less typing, better output. These tutorials where from release date but yes, most tutorials have the `apt-get` instead.